app/javascript/react/components/CoffeeShopShowContainer.js

```diff
@@ -28,21 +30,25 @@ const CoffeeShopShowContainer = (props) => {
   const addNewReview = async (formPayload) => {
     try {
       const reviewResponse = await fetch(`/api/v1/coffee_shops/${coffeeShopId}/reviews`, {
-        method: "POST",
+        method: 'POST',
         headers: {
-          "Content-Type": "application/json",
-          Accept: "application/json",
+          'Content-Type': 'application/json',
+          Accept: 'application/json'
         },
-        body: JSON.stringify(formPayload),
+        body: JSON.stringify(formPayload)
       });
       if (reviewResponse.ok) {
         const parsedReviewResponse = await reviewResponse.json();
 
         setCoffeeShop({
           ...coffeeShop,
-          reviews: [...coffeeShop.reviews, parsedReviewResponse.review],
+          reviews: [...coffeeShop.reviews, parsedReviewResponse.review]
         });
       }
+      if (reviewResponse.status === 401) {
```

Comment:
Looks like your controller is set up to send an error of unprocessable entity, which is a 422 error that isn't being checked for here. Additionally, the authenticate_user implementation will send a 400 status code but it won't send an error body with a message - you have the option to look for this status code and hard coding a message that the user must be logged in, or defining your own authenticate_user method in the API controller (the built in Devise authenticate user method isn't set up to work with a controller that returns JSON)

`authenticate_user!` essentially checks if there is a `current_user` in session or not, but won't provide a message. You could create your own reusable method in the `application_controller`, something like
```
  def authenticate_user
    if !user_signed_in?
      render json: {error: ["You need to be signed in first"]}
    end
  end
```

Reply:
The Devise method give an error message in the returned JSON